openmetadata-ui/src/main/resources/ui/src/components/Glossary/GlossaryTerms/tabs/AssetsTabs.component.tsx

```diff
@@ -111,7 +144,77 @@ const AssetsTabs = forwardRef(
           value: option.value,
         };
       });
-    }, [itemCount]);
+    }, [activeFilter, itemCount]);
+
+    const getAssetMenuCountBadge = useCallback(
+      (key: EntityType) =>
+        getCountBadge(
+          ASSET_SUB_MENU_FILTER.find((item) => item.key === key)
+            ?.children.map((item) => itemCount[item.value] ?? 0)
+            ?.reduce((acc, cv) => acc + cv, 0)
+        ),
+      [itemCount]
+    );
+
+    const getOptions = useCallback(
+      (
+        option: {
+          label: string;
+          key: EntityType | SearchIndex;
+          value?: EntityType;
+        },
+        isChildren?: boolean
+      ) => {
+        const assetCount = option.value && itemCount[option.value];
+
+        return {
+          label: isChildren ? (
+            <Space className="w-full d-flex justify-between">
+              <div className="d-flex items-center">
+                <span className="m-r-xs w-4 d-flex">
+                  {getEntityIcon(option.key)}
+                </span>
+
+                <Typography.Text
+                  className="w-32 text-color-inherit"
+                  ellipsis={{ tooltip: true }}>
+                  {option.label}
+                </Typography.Text>
+
+                {getCountBadge(assetCount)}
+              </div>
+              <Checkbox
+                checked={activeFilter.includes(option.key as SearchIndex)}
+              />
+            </Space>
+          ) : (
+            <div className="d-flex justify-between">
+              <span>{option.label}</span>
+              <span className="p-l-xs">
+                {getAssetMenuCountBadge(option.key as EntityType)}
+              </span>
+            </div>
+          ),
+          key: option.key,
+          value: option.key,
+          disabled: isChildren && !assetCount,
```

Comment:
Don't we need to disableSelection for header instead disabled the option itself

Reply:
@chirag-madlani  updated the changes to disable the header in case children does not have data!